nexus/src/app/vpc.rs

```diff
@@ -315,18 +329,425 @@ impl super::Nexus {
         vpc_name: &Name,
         params: &VpcFirewallRuleUpdateParams,
     ) -> UpdateResult<Vec<db::model::VpcFirewallRule>> {
-        let (.., authz_vpc) = LookupPath::new(opctx, &self.db_datastore)
-            .organization_name(organization_name)
-            .project_name(project_name)
-            .vpc_name(vpc_name)
-            .lookup_for(authz::Action::Modify)
-            .await?;
+        let (.., authz_vpc, db_vpc) =
+            LookupPath::new(opctx, &self.db_datastore)
+                .organization_name(organization_name)
+                .project_name(project_name)
+                .vpc_name(vpc_name)
+                .fetch_for(authz::Action::Modify)
+                .await?;
         let rules = db::model::VpcFirewallRule::vec_from_params(
             authz_vpc.id(),
             params.clone(),
         );
-        self.db_datastore
+        let rules = self
+            .db_datastore
             .vpc_update_firewall_rules(opctx, &authz_vpc, rules)
-            .await
+            .await?;
+        self.send_sled_agents_firewall_rules(opctx, &db_vpc, &rules).await?;
+        Ok(rules)
+    }
+
+    /// Customize the default firewall rules for a particular VPC.
```

Comment:
I think the comment could be a bit more explanatory. At first this appeared to just return the default rules, but I (think) it's actually inserting the real VPC name in place of the name `"default"`. If that's correct, that'd be very helpful to have in the comment.

Reply:
Correct, added in 171a6251.